compiler/rustc_codegen_cranelift/src/driver/aot.rs

```diff
@@ -129,9 +130,19 @@ fn module_codegen(
             }
             MonoItem::GlobalAsm(item_id) => {
                 let item = cx.tcx.hir().item(item_id);
-                if let rustc_hir::ItemKind::GlobalAsm(rustc_hir::GlobalAsm { asm }) = item.kind {
-                    cx.global_asm.push_str(&*asm.as_str());
-                    cx.global_asm.push_str("\n\n");
+                if let rustc_hir::ItemKind::GlobalAsm(asm) = item.kind {
+                    if !asm.options.contains(InlineAsmOptions::ATT_SYNTAX) {
+                        cx.global_asm.push_str("\n.intel_syntax\n");
```

Comment:
I think this should be `.intel_syntax noprefix`.

Reply:
Ah nice catch! LLVM's internal assembler uses `noprefix` by default but I forgot that GAS uses the opposite default.